anndata/readwrite/zarr.py

```diff
@@ -252,23 +254,17 @@ def read_zarr(store: Union[str, Path, MutableMapping, zarr.Group]) -> AnnData:
         else:  # Base case
             d[k] = read_attribute(f[k])
 
-    # Backwards compat
-    raw = {}
-    if "raw.var" in f:
-        raw["var"] = read_dataframe(f["raw.var"])  # Backwards compat
-    if "raw.varm" in f:
-        raw["varm"] = read_attribute(f["raw.varm"])
-    if "raw.X" in f:
-        raw["X"] = read_attribute(f["raw.X"])
-    if len(raw) > 0:
-        assert "raw" not in d
-        d["raw"] = raw
+    _read_legacy_raw_zarr(f, d.setdefault("raw", {}))
```

Comment:
Won’t it be skipped later if it’s empty?

Reply:
> If it is there, there would be a var

sure? Why?

Reply:
I was trying to think of what `skipped` means. I was thinking it could be when you're reading backed, (so there's an `X`, but it's not read).

Could you explain what you mean by "skipped"?

Reply:
`_init_as_actual` contains

```py
if raw and isinstance(raw, cabc.Mapping):
    ...
elif raw:  # is a Raw from another AnnData
    ...
self._raw = raw if raw else None
```

so if `raw == {}`, `self._raw` will be set to `None`.

Reply:
Alright. I'm a little worried about what's inside those branches, but if it works it's fine. Could you take a quick look over the tests to make sure any case this could effect is checked?